### PR TITLE
[libra-infallible] introduce a wrapper for std::sync::Rwlock

### PR DESCRIPTION
**Pull request by mimoo**
This change applies the same changes we did in https://github.com/libra/libra/pull/6442 but with RwLock

### CODE REVIEW

File: common/infallible/src/rwlock.rs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+}
+
+#[cfg(test)]
+mod tests {
+
+    use super::*;
+    use std::{sync::Arc, thread};
+
+    #[test]
+    fn test_libra_rwlock() {
```
Comment (davidiw) [inline common/infallible/src/rwlock.rs:46]: why add?

File: documentation/coding_guidelines.md [hunk 2]
```diff
@@ -181,7 +181,7 @@ Error handling suggestions follow the [Rust book guidance](https://doc.rust-lang
 
 *Panic*
```
Comment (davidiw) [inline documentation/coding_guidelines.md:182]: can you put in a blurb about infallible here?

**Review by davidiw — COMMENTED** [review]
Overall looks good, want to get the language right on the coding guidelines before accepting.

File: common/infallible/src/rwlock.rs (hunk 1 above)
Comment (mimoo) [reply common/infallible/src/rwlock.rs:46]: why add what?

File: documentation/coding_guidelines.md (hunk 2 above)
Comment (mimoo) [reply documentation/coding_guidelines.md:182]: yes sir!

Comment (mimoo) [reply documentation/coding_guidelines.md:187]: how about that?

**Comment by mimoo** [comment]
oh no, I have to rebase now ...

File: common/infallible/src/rwlock.rs (hunk 1 above)
Comment (davidiw) [reply common/infallible/src/rwlock.rs:46]: the test, why add a test that just verifies your wrapper api?

**Review by davidiw — APPROVED** [review]
/land

**Pull request closed without merging**